python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -329,6 +333,36 @@ public void visitAssignmentStatement(AssignmentStatement assignmentStatement) {
       });
   }
 
+  @Override
+  public void visitParameter(Parameter parameter) {
+    scan(parameter.typeAnnotation());
+    scan(parameter.defaultValue());
+    Optional.ofNullable(parameter.typeAnnotation())
+      .map(TrivialTypeInferenceVisitor::resolveTypeAnnotationType)
+      .ifPresent(type -> setTypeToName(parameter.name(), type));
+    scan(parameter.name());
+  }
+
+  private static PythonType resolveTypeAnnotationType(TypeAnnotation typeAnnotation) {
+    if (typeAnnotation.expression() instanceof Name name && name.typeV2() != PythonType.UNKNOWN) {
+      return new ObjectType(name.typeV2(), TypeSource.TYPE_HINT);
+    } else if (typeAnnotation.expression() instanceof SubscriptionExpression subscriptionExpression && subscriptionExpression.object().typeV2() != PythonType.UNKNOWN) {
```

Comment:
This is certainly out of scope for this PR, but we could have a test for syntax such as `X | Y` to represent union types as well.